server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3831,11 +3830,15 @@ private UserVm createVirtualMachine(DataCenter zone, ServiceOffering serviceOffe
         // check if account/domain is with in resource limits to create a new vm
         boolean isIso = Storage.ImageFormat.ISO == template.getFormat();
 
-        long size = configureCustomRootDiskSize(customParameters, template, hypervisorType, offering);
+        long volumesSize = isIso ? 0 : configureCustomRootDiskSize(customParameters, template, hypervisorType, offering);
```

Comment:
@sureshanaparti do we need to consider even ISO size for resource limit check done at line 3683? cc @DaanHoogland @rohityadavcloud @weizhouapache 
I've made the change as volume VMs deployed using ISOs will always have root volume created from the specified disk offering and ISO size shouldn't be considered

Reply:
Makes sense from a logical viewpoint but as an operator, will the ISO not also take its space on primary, i.e. will the user not have to be charged for that space?

Reply:
I'm not completely sure if ISO will be copied to primary store. But even if is and if the same ISO is used by 10 accounts to deploy a VM each, the operator will end up charging them all for ISO size.
Better for now, leave it as it is, too big a change for 4.16.1 :D I'm reverting it.

Reply:
ok, as long as it's clear and consistent we can always revisit. @sureshanaparti, agree ?

Reply:
agree @DaanHoogland